### ISSUE TITLE
fix: `allocate_loss` computes `deployed_ratio` with integer division, so it is always 0 or 1

### ISSUE THREAD

**Issue by sanmipaul** [astera-hq/Astera#1291]
In `contracts/tranche/src/repayment.rs`:

```rust
let deployed_ratio = pool.junior.deployed as u128 / junior_total as u128;
let available_loss = junior_loss - (junior_loss as u128 * deployed_ratio) as i128;
```

`deployed / total` in integer arithmetic is `0` whenever `deployed < total` and `1` when they are equal, so `available_loss` collapses to either the entire loss or zero. The intended proportional split between `deployed` and `available` never happens. The same pattern is repeated for the senior tranche a few lines below.

The ratio needs a scaling denominator (e.g. `deployed * BPS / total`) before being multiplied back out.
